ProjectVoyage/Cell/MessageCell.m

```diff
@@ -163,19 +170,31 @@ - (void) setTextView {
     textView_y = -3;
     autoresizing = UIViewAutoresizingFlexibleLeftMargin;
     
-    /* if sender is someone else {
-        textView_x = 20;
-        textView_y = -1;
-        autoresizing = UIViewAutoresizingFlexibleRightMargin;
-    */
-    
     _textView.autoresizingMask = autoresizing;
     _textView.frame = CGRectMake(textView_x, textView_y, textView_w, textView_h);
+    
+    if (_message.sender == MessageSenderSomeone) {
+        // Margins
+        CGFloat marginLeft = 5;
+        CGFloat marginRight = 2;
+        
+        // Position
+        CGFloat bubble_x;
+        CGFloat bubble_y = 0;
+        CGFloat bubble_width;
+        CGFloat bubble_height = _textView.frame.size.height + 8;
+        
+        bubble_x = marginRight;
+        bubble_width = _textView.frame.origin.x + _textView.frame.size.width + marginLeft;
+        
+        _bubbleImage.frame = CGRectMake(bubble_x, bubble_y, bubble_width, bubble_height);
+        _bubbleImage.autoresizingMask = _textView.autoresizingMask;
+    }
```

Comment:
I haven't vetted the code inside this if statement, (just copied from the view controller).

use the message cell to render your views based on the Message object. this is all this class should be responsible for. The view controller will be responsible for the logic